In [1]:
import pandas as pd

- Rank: A unique integer identifier that ranks the video games based on a specific criterion (likely sales). 
- Name: The name of the video game. 
- Platform: The gaming platform on which the video game was released (e.g., PS4, Xbox One, PC). 
- Year: The year the game was released. This column has some missing values, with 16,327 non-null entries out of 16,598. 
- Genre: The genre of the video game (e.g., Action, Sports, RPG). 
- Publisher: The company that published the video game. There are a few missing values, with 16,540 non-null entries. 
- NA_Sales: Sales in North America, measured in millions of units. 
- EU_Sales: Sales in Europe, measured in millions of units. 
- JP_Sales: Sales in Japan, measured in millions of units. 
- Other_Sales: Sales in regions other than North America, Europe, and Japan, measured in millions of units. 
- Global_Sales: The total global sales of the video game, measured in millions of units.

In [7]:
games=pd.read_csv("E:\pandas.DA\PANDAS_Learning\za\Online_game.csv")
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [8]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [10]:
games.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [11]:
games['Year'].unique()

array([2006., 1985., 2008., 2009., 1996., 1989., 1984., 2005., 1999.,
       2007., 2010., 2013., 2004., 1990., 1988., 2002., 2001., 2011.,
       1998., 2015., 2012., 2014., 1992., 1997., 1993., 1994., 1982.,
       2003., 1986., 2000.,   nan, 1995., 2016., 1991., 1981., 1987.,
       1980., 1983., 2020., 2017.])

In [15]:
games.dropna(subset='Year',inplace=True)

In [16]:
games['Year'].unique()

array([2006., 1985., 2008., 2009., 1996., 1989., 1984., 2005., 1999.,
       2007., 2010., 2013., 2004., 1990., 1988., 2002., 2001., 2011.,
       1998., 2015., 2012., 2014., 1992., 1997., 1993., 1994., 1982.,
       2003., 1986., 2000., 1995., 2016., 1991., 1981., 1987., 1980.,
       1983., 2020., 2017.])

In [19]:
games['Year']=games['Year'].astype('int64')

In [21]:
games['Year'].dtype()

TypeError: 'numpy.dtype[int64]' object is not callable

In [22]:
games.isnull().sum()

Rank             0
Name             0
Platform         0
Year             0
Genre            0
Publisher       36
NA_Sales         0
EU_Sales         0
JP_Sales         0
Other_Sales      0
Global_Sales     0
dtype: int64

# 1: Top 5 best sale games in the data 

In [23]:
top_5=games.sort_values(by='Global_Sales',ascending=False)

In [24]:
top_5.head(5)

Rank                      Name Platform  Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

# 2:Bottom 5 video game sales

In [25]:
bottom_5=games.sort_values(by='Global_Sales')
bottom_5.head(5)

Rank                                             Name Platform  Year  \
16597  16600                                 Spirits & Spells      GBA  2003   
16186  16189                                      BattleForge       PC  2009   
16185  16188           Tantei Jinguuji Saburo: Hai to Diamond      PSP  2009   
16184  16187  Fatal Fury: Battle Archives Volume 2 (JP sales)      PS2  2007   
16183  16186                      Sakura-Sou no Pet na Kanojo      PSP  2013   

           Genre         Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
16597   Platform           Wanadoo      0.01      0.00      0.00          0.0   
16186   Strategy   Electronic Arts      0.00      0.01      0.00          0.0   
16185  Adventure  Arc System Works      0.00      0.00      0.01          0.0   
16184   Fighting      SNK Playmore      0.00      0.00      0.01          0.0   
16183  Adventure    Kadokawa Games      0.00      0.00      0.01          0.0   

       Global_Sales  
16597          0.01  
16186          0.01  
16185          0.01  
16184          0.01  
16183          0.01

# 3: Which platform provides more game

In [26]:
games['Platform'].value_counts()

Platform
DS      2133
PS2     2127
PS3     1304
Wii     1290
X360    1235
PSP     1197
PS      1189
PC       943
GBA      811
XB       803
GC       542
3DS      500
PSV      412
PS4      336
N64      316
SNES     239
XOne     213
SAT      173
WiiU     143
2600     116
NES       98
GB        97
DC        52
GEN       27
NG        12
SCD        6
WS         6
3DO        3
TG16       2
GG         1
PCFX       1
Name: count, dtype: int64

- TAKEAWAY: DS platform produced 2163 games 

# 4: Games which is under the platform of Wii

In [27]:
games.loc[games['Platform']=='Wii']

Rank                                               Name Platform  \
0          1                                         Wii Sports      Wii   
2          3                                     Mario Kart Wii      Wii   
3          4                                  Wii Sports Resort      Wii   
7          8                                           Wii Play      Wii   
8          9                          New Super Mario Bros. Wii      Wii   
...      ...                                                ...      ...   
16517  16520                       Teenage Mutant Ninja Turtles      Wii   
16552  16555           Mahou Sensei Negima!? Neo-Pactio Fight!!      Wii   
16573  16576                                Mini Desktop Racing      Wii   
16574  16577  Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...      Wii   
16581  16584                                          Fit & Fun      Wii   

       Year     Genre                     Publisher  NA_Sales  EU_Sales  \
0      2006    Sports                      Nintendo     41.49     29.02   
2      2008    Racing                      Nintendo     15.85     12.88   
3      2009    Sports                      Nintendo     15.75     11.01   
7      2006      Misc                      Nintendo     14.03      9.20   
8      2009  Platform                      Nintendo     14.59      7.06   
...     ...       ...                           ...       ...       ...   
16517  2007    Action  Konami Digital Entertainment      0.00      0.01   
16552  2007  Fighting         Marvelous Interactive      0.00      0.00   
16573  2007    Racing                Popcorn Arcade      0.01      0.00   
16574  2008    Racing                   Takara Tomy      0.00      0.00   
16581  2011    Sports                       Unknown      0.00      0.01   

       JP_Sales  Other_Sales  Global_Sales  
0          3.77         8.46         82.74  
2          3.79         3.31         35.82  
3          3.28         2.96         33.00  
7          2.93         2.85         29.02  
8          4.70         2.26         28.62  
...         ...          ...           ...  
16517      0.00         0.00          0.01  
16552      0.01         0.00          0.01  
16573      0.00         0.00          0.01  
16574      0.01         0.00          0.01  
16581      0.00         0.00          0.01  

[1290 rows x 11 columns]

# 5: No:of sales in a year

In [28]:
games['Year'].value_counts()

Year
2009    1431
2008    1428
2010    1259
2007    1202
2011    1139
2006    1008
2005     941
2002     829
2003     775
2004     763
2012     657
2015     614
2014     582
2013     546
2001     482
1998     379
2000     349
2016     344
1999     338
1997     289
1996     263
1995     219
1994     121
1993      60
1981      46
1992      43
1991      41
1982      36
1986      21
1989      17
1983      17
1990      16
1987      16
1988      15
1985      14
1984      14
1980       9
2017       3
2020       1
Name: count, dtype: int64

- TAKEAWAY : In 2009 having a top sale of 1431 in online games

# 6:All  year sales 

In [29]:
games.pivot_table(index='Year',values=['NA_Sales','EU_Sales','JP_Sales','Other_Sales','Global_Sales'],aggfunc=sum)

EU_Sales  Global_Sales  JP_Sales  NA_Sales  Other_Sales
Year                                                         
1980      0.67         11.38      0.00     10.59         0.12
1981      1.96         35.77      0.00     33.40         0.32
1982      1.65         28.86      0.00     26.92         0.31
1983      0.80         16.79      8.10      7.76         0.14
1984      2.10         50.36     14.27     33.28         0.70
1985      4.74         53.94     14.56     33.73         0.92
1986      2.84         37.07     19.81     12.50         1.93
1987      1.41         21.74     11.63      8.46         0.20
1988      6.59         47.22     15.76     23.87         0.99
1989      8.44         73.45     18.36     45.15         1.50
1990      7.63         49.39     14.88     25.46         1.40
1991      3.95         32.23     14.78     12.76         0.74
1992     11.71         76.16     28.91     33.87         1.65
1993      4.65         45.98     25.33     15.12         0.89
1994     14.88         79.17     33.99     28.15         2.20
1995     14.90         88.11     45.75     24.82         2.64
1996     47.26        199.15     57.44     86.76         7.69
1997     48.32        200.98     48.87     94.75         9.13
1998     66.90        256.47     50.04    128.36        11.03
1999     62.67        251.27     52.34    126.06        10.05
2000     52.75        201.56     42.77     94.49        11.62
2001     94.89        331.47     39.86    173.98        22.76
2002    109.74        395.52     41.76    216.19        27.28
2003    103.81        357.85     34.20    193.59        26.01
2004    107.32        419.31     41.65    222.59        47.29
2005    121.94        459.94     54.28    242.61        40.58
2006    129.24        521.04     73.73    263.12        54.43
2007    160.50        611.13     60.29    312.05        77.60
2008    184.40        678.90     60.26    351.44        82.39
2009    191.59        667.30     61.89    338.85        74.77
2010    176.73        600.45     59.49    304.24        59.90
2011    167.44        515.99     53.04    241.06        54.39
2012    118.78        363.54     51.74    154.96        37.82
2013    125.80        368.11     47.59    154.77        39.82
2014    125.65        337.05     39.46    131.97        40.02
2015     97.71        264.44     33.72    102.82        30.01
2016     26.76         70.93     13.70     22.66         7.75
2017      0.00          0.05      0.05      0.00         0.00
2020      0.00          0.29      0.00      0.27         0.02

# 7: Games that having the  genre only sports and racing.There sales 

In [30]:
games.loc[(games['Genre']=='Sports')|(games['Genre']=='Racing') ]

Rank                                              Name Platform  Year  \
0          1                                        Wii Sports      Wii  2006   
2          3                                    Mario Kart Wii      Wii  2008   
3          4                                 Wii Sports Resort      Wii  2009   
11        12                                     Mario Kart DS       DS  2005   
13        14                                           Wii Fit      Wii  2007   
...      ...                                               ...      ...   ...   
16578  16581                     Outdoors Unleashed: Africa 3D      3DS  2011   
16579  16582                                 PGA European Tour      N64  2000   
16581  16584                                         Fit & Fun      Wii  2011   
16587  16590                          Mezase!! Tsuri Master DS       DS  2009   
16595  16598  SCORE International Baja 1000: The Official Game      PS2  2008   

        Genre    Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0      Sports     Nintendo     41.49     29.02      3.77         8.46   
2      Racing     Nintendo     15.85     12.88      3.79         3.31   
3      Sports     Nintendo     15.75     11.01      3.28         2.96   
11     Racing     Nintendo      9.81      7.57      4.13         1.92   
13     Sports     Nintendo      8.94      8.03      3.60         2.15   
...       ...          ...       ...       ...       ...          ...   
16578  Sports      Mastiff      0.01      0.00      0.00         0.00   
16579  Sports   Infogrames      0.01      0.00      0.00         0.00   
16581  Sports      Unknown      0.00      0.01      0.00         0.00   
16587  Sports  Hudson Soft      0.00      0.00      0.01         0.00   
16595  Racing   Activision      0.00      0.00      0.00         0.00   

       Global_Sales  
0             82.74  
2             35.82  
3             33.00  
11            23.42  
13            22.72  
...             ...  
16578          0.01  
16579          0.01  
16581          0.01  
16587          0.01  
16595          0.01  

[3530 rows x 11 columns]

# 8: Which country have more sales

In [31]:
games.pivot_table(index='Year',values=['NA_Sales','EU_Sales','JP_Sales','Other_Sales'],aggfunc=sum,margins=True,margins_name='Total')

EU_Sales  JP_Sales  NA_Sales  Other_Sales
Year                                            
1980       0.67      0.00     10.59         0.12
1981       1.96      0.00     33.40         0.32
1982       1.65      0.00     26.92         0.31
1983       0.80      8.10      7.76         0.14
1984       2.10     14.27     33.28         0.70
1985       4.74     14.56     33.73         0.92
1986       2.84     19.81     12.50         1.93
1987       1.41     11.63      8.46         0.20
1988       6.59     15.76     23.87         0.99
1989       8.44     18.36     45.15         1.50
1990       7.63     14.88     25.46         1.40
1991       3.95     14.78     12.76         0.74
1992      11.71     28.91     33.87         1.65
1993       4.65     25.33     15.12         0.89
1994      14.88     33.99     28.15         2.20
1995      14.90     45.75     24.82         2.64
1996      47.26     57.44     86.76         7.69
1997      48.32     48.87     94.75         9.13
1998      66.90     50.04    128.36        11.03
1999      62.67     52.34    126.06        10.05
2000      52.75     42.77     94.49        11.62
2001      94.89     39.86    173.98        22.76
2002     109.74     41.76    216.19        27.28
2003     103.81     34.20    193.59        26.01
2004     107.32     41.65    222.59        47.29
2005     121.94     54.28    242.61        40.58
2006     129.24     73.73    263.12        54.43
2007     160.50     60.29    312.05        77.60
2008     184.40     60.26    351.44        82.39
2009     191.59     61.89    338.85        74.77
2010     176.73     59.49    304.24        59.90
2011     167.44     53.04    241.06        54.39
2012     118.78     51.74    154.96        37.82
2013     125.80     47.59    154.77        39.82
2014     125.65     39.46    131.97        40.02
2015      97.71     33.72    102.82        30.01
2016      26.76     13.70     22.66         7.75
2017       0.00      0.05      0.00         0.00
2020       0.00      0.00      0.27         0.02
Total   2409.12   1284.30   4333.43       789.01

- TAKEAWAY: North America having most sales 

# 9: Which of the games that have more than 10 sales in Japan

In [32]:
games.loc[games['JP_Sales']>10]

Rank                      Name Platform  Year         Genre Publisher  \
4     5  Pokemon Red/Pokemon Blue       GB  1996  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
4     11.27      8.89     10.22          1.0         31.37

 - TAKEAWAY : Only Pokemon Red/Pokemon Blue game which having a sales of 10.22 in Japan 

# 10: Want to get the details that having the platform Wii and publisher Nintendo

In [33]:
games[(games['Platform']=='Wii')&(games['Publisher']=='Nintendo')]

Rank                                         Name Platform  Year  \
0          1                                   Wii Sports      Wii  2006   
2          3                               Mario Kart Wii      Wii  2008   
3          4                            Wii Sports Resort      Wii  2009   
7          8                                     Wii Play      Wii  2006   
8          9                    New Super Mario Bros. Wii      Wii  2009   
...      ...                                          ...      ...   ...   
13314  13316                     Wii de Asobu Chibi-Robo!      Wii  2009   
13523  13525  Minna ga Shuyaku no NHK Kouhaku Quiz Kassen      Wii  2009   
14376  14379  Eyeshield 21: Field Saikyou no Senshi Tachi      Wii  2007   
15991  15994                      Donkey Kong Jungle Beat      Wii  2008   
16473  16476                              Captain Rainbow      Wii  2008   

              Genre Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0            Sports  Nintendo     41.49     29.02      3.77         8.46   
2            Racing  Nintendo     15.85     12.88      3.79         3.31   
3            Sports  Nintendo     15.75     11.01      3.28         2.96   
7              Misc  Nintendo     14.03      9.20      2.93         2.85   
8          Platform  Nintendo     14.59      7.06      4.70         2.26   
...             ...       ...       ...       ...       ...          ...   
13314     Adventure  Nintendo      0.00      0.00      0.05         0.00   
13523          Misc  Nintendo      0.00      0.00      0.04         0.00   
14376  Role-Playing  Nintendo      0.00      0.00      0.03         0.00   
15991      Platform  Nintendo      0.00      0.01      0.00         0.00   
16473     Adventure  Nintendo      0.00      0.00      0.01         0.00   

       Global_Sales  
0             82.74  
2             35.82  
3             33.00  
7             29.02  
8             28.62  
...             ...  
13314          0.05  
13523          0.04  
14376          0.03  
15991          0.01  
16473          0.01  

[81 rows x 11 columns]

 - TAKEAWAY: There were 85 online games published by Nintendo in Wii platform

# 11:Top 10 games and corresponding sales .Who is publisher?

In [36]:
games.loc[0:10,['Name','Publisher','NA_Sales','EU_Sales','JP_Sales','Other_Sales']]

Name Publisher  NA_Sales  EU_Sales  JP_Sales  \
0                  Wii Sports  Nintendo     41.49     29.02      3.77   
1           Super Mario Bros.  Nintendo     29.08      3.58      6.81   
2              Mario Kart Wii  Nintendo     15.85     12.88      3.79   
3           Wii Sports Resort  Nintendo     15.75     11.01      3.28   
4    Pokemon Red/Pokemon Blue  Nintendo     11.27      8.89     10.22   
5                      Tetris  Nintendo     23.20      2.26      4.22   
6       New Super Mario Bros.  Nintendo     11.38      9.23      6.50   
7                    Wii Play  Nintendo     14.03      9.20      2.93   
8   New Super Mario Bros. Wii  Nintendo     14.59      7.06      4.70   
9                   Duck Hunt  Nintendo     26.93      0.63      0.28   
10                 Nintendogs  Nintendo      9.07     11.00      1.93   

    Other_Sales  
0          8.46  
1          0.77  
2          3.31  
3          2.96  
4          1.00  
5          0.58  
6          2.90  
7          2.85  
8          2.26  
9          0.47  
10         2.75

- TAKEAWAY: Nintendo is best publisher who published top 10 rank games 

# 12:In which genre the publisher 'Nintendo' published the most

In [44]:
genre_and_nintendo=games.loc[games['Publisher']=='Nintendo'].loc[:,['Genre','Publisher']]
genre_and_nintendo

Genre Publisher
0            Sports  Nintendo
1          Platform  Nintendo
2            Racing  Nintendo
3            Sports  Nintendo
4      Role-Playing  Nintendo
...             ...       ...
16269        Action  Nintendo
16357        Puzzle  Nintendo
16456          Misc  Nintendo
16473     Adventure  Nintendo
16542          Misc  Nintendo

[696 rows x 2 columns]

In [45]:
genre_and_nintendo.value_counts()

Genre         Publisher
Platform      Nintendo     111
Role-Playing  Nintendo     105
Misc          Nintendo     100
Action        Nintendo      78
Puzzle        Nintendo      74
Sports        Nintendo      55
Racing        Nintendo      37
Adventure     Nintendo      35
Strategy      Nintendo      31
Simulation    Nintendo      28
Shooter       Nintendo      24
Fighting      Nintendo      18
Name: count, dtype: int64

- TAKEAWAY: In Platform genre Nintendo published the most about 111 games

# 13: Which is best game in 2007 and its rank in Depends on the whole year game sales data,global sales(in million units)

In [55]:
games.loc[games['Year']==2007].loc[:,['Rank','Name','Year','Global_Sales']]

Rank                                               Name  Year  \
13        14                                            Wii Fit  2007   
43        44                                             Halo 3  2007   
48        49                                 Super Mario Galaxy  2007   
70        71                     Call of Duty 4: Modern Warfare  2007   
74        75                                     Mario Party DS  2007   
...      ...                                                ...   ...   
16552  16555           Mahou Sensei Negima!? Neo-Pactio Fight!!  2007   
16558  16561                          Pro Evolution Soccer 2008  2007   
16564  16567      Original Frisbee Disc Sports: Ultimate & Golf  2007   
16573  16576                                Mini Desktop Racing  2007   
16590  16593  Eiyuu Densetsu: Sora no Kiseki Material Collec...  2007   

       Global_Sales  
13            22.72  
43            12.14  
48            11.52  
70             9.32  
74             9.02  
...             ...  
16552          0.01  
16558          0.01  
16564          0.01  
16573          0.01  
16590          0.01  

[1202 rows x 4 columns]

- TAKEAWAY:There are 1202 video games in 2007 in which 'Will Fit' having a top sales of 22.72 (million units).Its rank according to other year video games were 14

# 14:video games that having a global sales of 0.01

In [63]:
less_sales=games.loc[games['Global_Sales']==0.01].loc[:,['Name','Platform','Publisher']]
less_sales

Name Platform  \
15980                                              Turok       PC   
15981                     Coven and Labyrinth of Refrain      PSV   
15982  Super Battle For Money Sentouchuu: Kyuukyoku n...      3DS   
15983                                   Dragon Zakura DS       DS   
15984                             Chameleon: To Dye For!       DS   
...                                                  ...      ...   
16593                 Woody Woodpecker in Crazy Castle 5      GBA   
16594                      Men in Black II: Alien Escape       GC   
16595   SCORE International Baja 1000: The Official Game      PS2   
16596                                         Know How 2       DS   
16597                                   Spirits & Spells      GBA   

                  Publisher  
15980            Touchstone  
15981  Nippon Ichi Software  
15982    Namco Bandai Games  
15983       Electronic Arts  
15984             505 Games  
...                     ...  
16593                 Kemco  
16594            Infogrames  
16595            Activision  
16596              7G//AMES  
16597               Wanadoo  

[605 rows x 3 columns]

- TAKEAWAY: There were 605 video games which having 0.01 Global sales 

# 15: global sales of 0.01 which platform provides more games

In [66]:
less_sales['Platform'].value_counts()

Platform
PC      102
PSP      91
DS       88
PS2      67
PSV      47
X360     32
GBA      31
Wii      29
PS3      27
3DS      23
XB       15
XOne     14
PS4      11
GC       11
PS        8
WiiU      6
N64       2
SNES      1
Name: count, dtype: int64

- TAKEAWAY: PC Platform produces 102 games 

# 16: Which are the video games that does not have sale in North America

In [68]:
no_sales_NA=games.loc[games['NA_Sales']==0.00].loc[:,['Name','NA_Sales']]
no_sales_NA

Name  NA_Sales
214                             Monster Hunter Freedom 3       0.0
338                                    Friend Collection       0.0
383                                     Monster Hunter 4       0.0
402    English Training: Have Fun Improving Your Skills!       0.0
426                 Dragon Quest VI: Maboroshi no Daichi       0.0
...                                                  ...       ...
16587                           Mezase!! Tsuri Master DS       0.0
16589  Chou Ezaru wa Akai Hana: Koi wa Tsuki ni Shiru...       0.0
16590  Eiyuu Densetsu: Sora no Kiseki Material Collec...       0.0
16595   SCORE International Baja 1000: The Official Game       0.0
16596                                         Know How 2       0.0

[4439 rows x 2 columns]

In [70]:
no_sales_NA.nunique()

Name        4030
NA_Sales       1
dtype: int64

- TAKEAWAY: There were 4030 video games that have no sales in North America

# 17: Whether GB Platform is in top 10 video games 

In [73]:
games.loc[games['Platform']=='GB'].loc[0:9,['Name','Platform']]

Name Platform
4  Pokemon Red/Pokemon Blue       GB
5                    Tetris       GB

- TAKEAWAY: There were two video games in the platform of GB in top 10 they were in rank 4,rank 5

# 18:In top 10 which video game genre is going more

In [76]:
Top_10_genre=games.loc[0:9,['Rank','Name','Genre']]
Top_10_genre

Rank                       Name         Genre
0     1                 Wii Sports        Sports
1     2          Super Mario Bros.      Platform
2     3             Mario Kart Wii        Racing
3     4          Wii Sports Resort        Sports
4     5   Pokemon Red/Pokemon Blue  Role-Playing
5     6                     Tetris        Puzzle
6     7      New Super Mario Bros.      Platform
7     8                   Wii Play          Misc
8     9  New Super Mario Bros. Wii      Platform
9    10                  Duck Hunt       Shooter

In [78]:
Top_10_genre['Genre'].value_counts()

Genre
Platform        3
Sports          2
Racing          1
Role-Playing    1
Puzzle          1
Misc            1
Shooter         1
Name: count, dtype: int64

- TAKEAWAY: UnderPlatform genre having 3 video games in top 10 video games 

# 19:Who published the most video games 

In [79]:
games['Publisher'].value_counts()

Publisher
Electronic Arts                 1339
Activision                       966
Namco Bandai Games               928
Ubisoft                          918
Konami Digital Entertainment     823
                                ... 
Detn8 Games                        1
Pow                                1
Navarre Corp                       1
MediaQuest                         1
UIG Entertainment                  1
Name: count, Length: 576, dtype: int64

- TAKEAWAY: Electronic Arts published 1339 video games 

# 20: which of the video games,platform and genre published by 'Wanadoo'

In [84]:
games.loc[games['Publisher']=='Wanadoo'].loc[:,['Name','Genre']]

Name     Genre
4238   Rygar: The Legendary Adventure    Action
9016                      Fatal Frame    Action
15675                Spirits & Spells  Platform
15734             Top Gear RPM Tuning    Racing
16597                Spirits & Spells  Platform

- TAKEAWAY: Wanadoo published 5 video games which is ranked higher that 1000  